tests/neg/local-implicit-function.scala

```diff
@@ -0,0 +1,45 @@
+object Test {
+  type using[A,E] = implicit local E => A
+
+  case class Incrementor(i1: Int) {
+    def incr(i2: Int) = i1 + i2
+  }
+
+  def incr(i: Int): Int using Incrementor = implicitly[Incrementor].incr(i)
```

Comment:
Why does `implicitly` work at all? isn't it defined as
```
def implicitly[A](implicit a: A): A = a
```
?
That would require `a` to be first class, doesn't it?

Reply:
Yes, `implicitly` breaks `local` somehow.

I investigated this some time ago and my suspicion is that `@forceInline` in the definition of `implicitly` has something to do with that.

```scala
@forceInline final def implicitly[T](implicit ev: T): T = ev
```

So I tested it now:

```scala
object Test {

  implicit local val foo: Int = 42
  @forceInline final def explicitly[T](         ev: T): T = ev
  @forceInline final def implicitly[T](implicit ev: T): T = ev
  local val bar  = explicitly(foo)
  local val cafe = implicitly[Int]

}
```

This works. So the code seems to get inlined before our phase.

Reply:
Maybe that's a feature and not a bug :)

Reply:
(as long as `val bar = implicitly[Int]` doesn't type check)

Reply:
Brilliant!

```scala
  local val x = 42
  var y: Int = _
  @forceInline final def tester(ev: Int): Unit = y = ev
  tester(x)
  //  |  @forceInline final def tester(ev: Int): Unit = y = ev
  //  |                                                     ^^
  //  |                            Found local value x in first class position.
  //  | This location is in code that was inlined at examples/test.scala:49
```